src/globus_sdk/experimental/globus_app/errors.py

```diff
@@ -50,7 +50,9 @@ def __init__(self, expires_at_seconds: int):
 class UnmetScopeRequirementsError(TokenValidationError):
     """The token stored for a given resource server is missing required scopes."""
 
-    def __init__(self, message: str, scope_requirements: dict[str, list[Scope]]):
+    def __init__(
+        self, message: str, scope_requirements: dict[str, list[Scope]]
```

Comment:
Did this need to be a mapping as well?

Reply:
I intentionally omitted that based on three facts (not so much "reasons" as considerations or factors).

1. It's an error constructor.
    Error construction isn't the typical user-facing interface (usually it's consumption of built instances). So there isn't as much to be gained in this case.

2. Changing this would mean that we need to make a runtime change.
    It would be easy enough to accept a `Mapping[str, Sequence[Scope]]` and then convert it with `{k: list(v) for k, v ...`, but I'm not convinced in this case.

3. The declared type does not include a union in the invariant types.
    These sorts of changes are much more significant where a union is used (since `list[int | str]` will reject `list[int]`, etc). Since there's no union here, there aren't many cases in which a type which looks like a subtype would be rejected. If you subclass `Scope` you can get there, but that's about it.